src/imgui/ImGuiUtils.cc

```diff
@@ -283,6 +284,160 @@ std::string getShortCutForCommand(const HotKey& hotkey, std::string_view command
 	return "";
 }
 
+// Adapted from ImGui_ImplSDL2_KeycodeToImGuiKey
+static int ImGuiKeyToSDL2_KeyCode(ImGuiKey key)
```

Comment:
I agree we need something like this.

It's unfortunate that this duplicates some information that's already present in `imgui_impl_sdl2`. That one contains the mapping from SDL to ImGui, but we also need the other direction (but in essence that's the same information, just not in a form that's easily accessible to us).

In fact the `ImGui_ImplSDL2_KeycodeToImGuiKey()` function itself was also not accessible, so you had to patch two Dear ImGui files. That's possible of course (and better than duplicating), but we should still try to avoid this (because it does increase the effort of upgrading to a newer Dear ImGui version).

The combination of the above gave me the following idea:
* Given that we have to duplicate this mapping on our side anyway. Why not use this information to generate the mapping functions for both directions.
* That way we don't have patch any Dear ImGui files.
* And both mappings will be consistent-by-construction. E.g. if a newer Dear ImGui version adds a new mapping, and we forget to update our version, then the mapping will be missing from both directions (and not present in one function but not the other). I think it's better that both conversions are consistent than one conversion being capable of more than the other (but of course we should try to keep our mapping in sync with Dear ImGui).

I created a prototype here:  https://godbolt.org/z/P415a1Mse
Summary:
* The code contains only one table to define the relation between corresponding ImGui and SDL keys.
* But this table does NOT end up in the generated executable.
* Instead from this table (at-compile-time) we generate two other tables (that do end up in the executable).
* These two tables are respectively sorted on the imgui or the sdl field.
* And in sorted tables we can do a binary search.